chromium_src/third_party/blink/renderer/core/dom/document.cc

```diff
@@ -22,6 +23,33 @@
 
 namespace {
 
+//  Returns the eTLD+1 for the top level frame the document is in.
+//
+//  Returns the eTLD+1 (effective registrable domain) for the top level
+//  frame that the given document is in. This includes frames that
+//  are disconnect, remote or local to the top level frame.
+//
+//  If the top level frame is not discoverable, an empty string is returned.
```

Comment:
when does this happen?

Reply:
We don't know all the cases, but I updated the comment to include an example